### PR TITLE
Add #warning directive

### LINKED ISSUE #169: Support for #warning preprocessor directive
Just as it says on the tin, I'd like to see support for Pawn's #warning preprocessor directive implemented in a future version of Sourcepawn, as an alternative to the compilation-halting #error directive. This is obviously incredibly niche and would only really find use for a very small selection of those of us creating includes for use in others' plugins.

(I'm also not sure how to tag this as a feature request or enhancement, or if someone else has to do that for me.)

### PR DESCRIPTION
**Pull request by Headline**
Here's my test code along with compiler outputs. The first is how the current `#error` directive functions, and the second is the new `#warning` directive
```cpp
/*
C:\Users\Micha\Desktop\New folder\sourcepawn\build\compiler\spcomp>spcomp test.sp
SourcePawn Compiler 1.10
Copyright (c) 1997-2006 ITB CompuPhase
Copyright (c) 2004-2017 AlliedModders LLC

test.sp(2) : fatal error 194: user error: testing 1 2 3


Compilation aborted.Code size:                0 bytes
Data size:                0 bytes
Stack/heap size:      16384 bytes
Total requirements:   16384 bytes

*/
#error testing 1 2 3 

int main()
{

}
```
```cpp
/*
C:\Users\Micha\Desktop\New folder\sourcepawn\build\compiler\spcomp>spcomp test.sp
SourcePawn Compiler 1.10
Copyright (c) 1997-2006 ITB CompuPhase
Copyright (c) 2004-2017 AlliedModders LLC

test.sp(2) : warning 241: user warning: testing 1 2 3

Code size:               24 bytes
Data size:                0 bytes
Stack/heap size:      16384 bytes
Total requirements:   16408 bytes

1 Warning.
*/
#warning testing 1 2 3 

int main()
{

}
```
Fixes #169

### CODE REVIEW

File: compiler/sc2.cpp [hunk 1]
```diff
@@ -1946,8 +1952,8 @@ const char *sc_tokens[] = {
          "while",
          "with",
          "#assert", "#define", "#else", "#elseif", "#endif", "#endinput",
-         "#endscript", "#error", "#file", "#if", "#include", "#line", "#pragma",
-         "#tryinclude", "#undef",
+         "#endscript", "#error", "#warning", "#file", "#if", "#include",
+		 "#line", "#pragma", "#tryinclude", "#undef",
```
Comment (asherkin) [inline compiler/sc2.cpp]: Mystery tabs.

File: compiler/sc5-in.scp [hunk 2]
```diff
@@ -274,4 +274,5 @@ static const char *warnmsg[] = {
 /*238*/  "'%s:' is an illegal cast; use view_as<%s>(expression)\n",
 /*239*/  "'%s' is an illegal tag; use %s as a type\n",
 /*240*/  "'%s:' is an old-style tag operation; use view_as<%s>(expression) instead\n",
+/*241*/  "user warning: %s\n",
```
Comment (asherkin) [inline compiler/sc5-in.scp]: Can we use one of the unused ones?

Comment (Headline) [reply compiler/sc5-in.scp]: I don't see why not, is it generally preferred to use the lowest unused spot?

Comment (asherkin) [reply compiler/sc5-in.scp]: Ya, preferred, but isn't super important.

Comment (KyleSanderson) [reply compiler/sc5-in.scp]: Patching holes is definitely a nice to have.

Comment (Headline) [reply compiler/sc5-in.scp]: Will do. I’ll probably commit in the coming days

**Comment by Headline** [comment]
Alright. Let me know if there's anything else.

@dvander

**Pull request merged**